packages/sdk/src/types/runtime/actor.ts

```diff
@@ -80,20 +81,21 @@ export interface ActorSet {
  */
 export class Actor implements ActorLike, Patchable<ActorLike> {
     // tslint:disable:variable-name
-    private _internal: InternalActor;
+    private _internal = new InternalActor(this);
     /** @hidden */
     public get internal() { return this._internal; }
 
-    private _emitter: events.EventEmitter;
+    private _emitter = new events.EventEmitter();
     /** @hidden */
     public get emitter() { return this._emitter; }
 
     private _name: string;
     private _tag: string;
-    private _parentId: string;
+    private _exclusiveToUser: User;

```

Comment:
>User [](start = 30, length = 4)

Let's store this as a string. It should be ok for actors to be assigned to users that aren't joined yet.

Reply:
That's a good design idea, but user IDs are useless to users. I'd much rather give them a User object if I can. What if it's stored as an ID, then the getter resolves it to a User?

---
In reply to: [272676558](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/293#discussion_r272676558) [](ancestors = 272676558)

Reply:
I think it needs to return a string so that it will function correctly when the user isn't in the space. The value returned by the getter should be consistent whether or not the user is in the space.

---
In reply to: [272693571](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/293#discussion_r272693571) [](ancestors = 272693571,272676558)

Reply:
I've changed this to a plain string. Union types are a pain to handle correctly, so at least a string can be handled consistently.

---
In reply to: [272696386](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/293#discussion_r272696386) [](ancestors = 272696386,272693571,272676558)